src/Services/AuthService.cs

```diff
@@ -1,89 +1,118 @@
-
-        return newUser;
-    }
-
-    public async Task RequestResetPasswordAsync(RequestResetPassword request)
-    {
-        var user = await _userRepository.GetByEmailAsync(request.Email);
-        user.ThrowIfNull("User not found");
-
-        var resetToken = _jwtService.GenerateToken(user!.Id, user!.Email, "resetPassword", TimeSpan.FromHours(1));
-
-        await _emailSender.SendEmail(user.Email, "Reset Your Password", $"Copy the token to  reset your password: {resetToken}");
-    }
-
-    public async Task ResetPasswordAsync(ResetPassword reset)
-    {
-        var email = _jwtService.ValidateToken(reset.Token, "resetPassword");
-
-        var user = await _userRepository.GetByEmailAsync(email);
-        user.ThrowIfNull("User not found");
-
-        var hashedPassword = _passwordHasher.HashPassword(reset.Password);
-        user!.PasswordHash = hashedPassword;
-
-        await _userRepository.UpdateAsync(user);
-    }
-}
+using Domain.DTOs;
+using Domain.Entities;
+
+using ExpenseTrackerGroup3.Domain.DTOs;
+using ExpenseTrackerGroup3.Exceptions;
+using ExpenseTrackerGroup3.Repositories.Interfaces;
+using ExpenseTrackerGroup3.Services.Interfaces;
+using ExpenseTrackerGroup3.Utils.EmailSender;
+using ExpenseTrackerGroup3.Utils.Exception;
+using ExpenseTrackerGroup3.Utils.Hasher.Interfaces;
+using ExpenseTrackerGroup3.Utils.Jwt.Interfaces;
+using FluentValidation;
+
+namespace ExpenseTrackerGroup3.Services;
+
+public class AuthService : IAuthService
+{
+    private readonly IUserRepository _userRepository;
+    private readonly IPasswordHasher _passwordHasher;
+    private readonly IJwtService _jwtService;
+    private readonly IEmailSender _emailSender;
+    private readonly IValidator<LoginRequest> _loginValidator;
+    private readonly IValidator<CreateUser> _createValidator;
+    private readonly IValidator<RequestResetPassword> _requestResetValidator;
+    private readonly IValidator<ResetPassword> _resetValidator;
+
+    public AuthService(
+        IUserRepository userRepository, 
+        IPasswordHasher passwordHasher, 
+        IJwtService jwtService, 
+        IEmailSender emailSender, 
+        IValidator<LoginRequest> loginValidator, 
+        IValidator<CreateUser> createValidator, 
+        IValidator<RequestResetPassword> requestResetValidator, 
+        IValidator<ResetPassword> resetValidator)
+    {
+        _userRepository = userRepository;
+        _passwordHasher = passwordHasher;
+        _jwtService = jwtService;
+        _emailSender = emailSender;
+        _loginValidator = loginValidator;
+        _createValidator = createValidator;
+        _requestResetValidator = requestResetValidator; 
+        _resetValidator = resetValidator;
+    }
+
+    public async Task<LoginResponse> LoginUserAsync(string email, string password)
+    {
+        var validationResult = await _loginValidator.ValidateAsync( new LoginRequest(email, password) );
+        validationResult.ThrowIfValidationFailed();

```

Comment:
Same thing with this, you're only modifying theses 2 lines, but, the entire class is modified.